app/src/main/java/blockeq/com/stellarwallet/activities/PinActivity.kt

```diff
@@ -54,81 +55,85 @@ class PinActivity : BaseActivity(), PinLockListener {
     }
 
     override fun onComplete(pin: String) {
-        when (pinViewState!!.type) {
```

Comment:
Not sure why the diff is so fragmented, all i did was wrap the logic in a handler and then call the handler.